util/spot.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"net/http"
+
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+// SpotHostWillTerminateSoon returns true if the EC2 spot host it is running on will terminate soon.
+func SpotHostWillTerminateSoon() bool {
+	const url = "http://169.254.169.254/latest/meta-data/spot/termination-time"
+	c := GetHTTPClient()
+	defer PutHTTPClient(c)
+	resp, err := c.Get(url)
```

Comment:
would there be an advantage to using the [SDK's version](https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/#EC2Metadata.GetMetadata)?

Reply:
My thought was that if in the near future we want to separate the agent from the app code, one of the biggest dependencies will be on the SDK. Since the agent only needs to GET a URL, it's reasonable to avoid using the SDK here to avoid the giant dependency. It does feel a little bit like premature optimization, but I don't expect us to do anything else with the SDK in the agent.

Reply:
but isn't the SDK external to both the agent and app? Are you saying that in the world where the agent has been severed from the app it wouldn't make sense to vendor the SDK for just one thing?

Reply:
yeah

Reply:
sgtm